### PR TITLE
Add Type 11(Attachment) to SlashCommandOptionType

### PR DESCRIPTION
**Pull request by n-Ultima**
## Description

Added the `Attachment` option to SlashCommandOptionType.

## Checklist

[//]: # "Put an x in [ ] to check the checkbox, e.g. [x]"

- [ ] I discussed this PR with the maintainer(s) prior to opening it.
- [x] I read the [contributing guidelines](https://github.com/Quahu/Disqord/blob/master/.github/CONTRIBUTING.md).
- [x] I tested the changes in this PR.

### CODE REVIEW

**Comment by n-Ultima** [comment]
Worth noting that at the time of writing, this is still pending in the discord docs, and the frontend is non existent.

**Review by AnotherZane — APPROVED** [review]

File: src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs [hunk 1]
```diff
@@ -88,6 +88,21 @@ public class TransientApplicationCommandInteractionEntities : TransientClientEnt
...
         private IReadOnlyDictionary<Snowflake, IMessage> _messages;
+        
+        /// <inheritdoc /> 
+        public IReadOnlyDictionary<Snowflake, IAttachment> Attachments
+        {
+            get
+            {
+                if (!Model.Attachments.HasValue)
+                    return ReadOnlyDictionary<Snowflake, IAttachment>.Empty;
+                return _attachments ??= Model.Attachments.Value.ToReadOnlyDictionary(
```
Comment (Quahu) [inline src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs:99]: Add a newline between the returns.

File: src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs [hunk 2]
```diff
@@ -88,6 +88,21 @@ public class TransientApplicationCommandInteractionEntities : TransientClientEnt
...
+                    kvp => kvp.Key,
+                    kvp => new TransientAttachment(kvp.Value) as IAttachment);
+            }
+        }
+
```
Comment (Quahu) [inline src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs:104]: Remove this newline.

**Review by Quahu — CHANGES_REQUESTED** [review]

File: src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs (hunk 1 above)
Comment (n-Ultima) [reply src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs]: Done in latest commit.

File: src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs (hunk 2 above)
Comment (n-Ultima) [reply src/Disqord.Core/Entities/Shared/Transient/Interactions/Commands/TransientApplicationCommandInteractionEntities.cs]: Done in latest commit.

**Pull request merged**